Simple beam hardening correction

minnititino opened the pull request:
I would like to introduce the following function:

- Simple beam hardening correction ''Simple BHC";


The Simple BHC making use of the following algorithm:
s = -ln(I/I0)
s’= s + a0 s^2 +a1 s^3+ a2 s^4 + a3 s^5

where a0, a1, a2 and a3 are parameters.

minnititino committed:
Create pproc_corr

minnititino committed:
Update pproc_corr

minnititino committed:
Update preproc.py

minnititino committed:
Delete pproc_corr

minnititino committed:
Update preproc.py

minnititino committed:
Create pproc_corr

minnititino committed:
Delete pproc_corr

dmici commented:
Dear Tino,

thanks for this new function. 

I would advise you to change the default values of the variables a0, a1, a2, a3. In fact for a0=0., a1=0., a2=0., a3=0. the correction is not performed.

Please modify the `__author__` variable at the beginning of the script in this way:

`__author__  =  "Davide Micieli, Triestino Minniti"	`

and at the next line append the name of the BHC function to the list  `__all__`.

minnititino committed:
Update preproc.py

minnititino committed:
zero_clipping_value function

Please consider to introduce the zero_clipping_value function that it is used to prevent values 
close to (or below) zero in the normalised images that might introduce new artefacts.

This function could be used also in the reconstructed images to prevent negative values of the calculated linear attenuation coefficients.

minnititino commented:
Dear Davide,

I have made the corrections that you have suggested and added the zero_clipping_value function that it is used to prevent values close to (or below) zero in the normalised images that might introduce new artefacts.

This function could be used also in the reconstructed images to prevent negative values of the calculated linear attenuation coefficients.

Tino.
